test/test_sparse.py

```diff
@@ -2177,6 +2177,20 @@ def test_factory_empty_indices(self):
         expected_indices = torch.empty((4, 0), dtype=torch.long, device=device)
         self.assertEqual(tensor._indices(), expected_indices)
 
+    def test_sparse_abs(self):
+        device = 'cuda' if self.is_cuda else 'cpu'
```

Comment:
For example, the generic device type test framework will automatically instantiate one test per available device type.